src/schema/models/schemaManager.ts

```diff
@@ -19,12 +23,31 @@ export class SchemaManager {
     return this.schemas.find((schema) => schema.name === name);
   }
 
-  public map(name: string, tags: Tags): Tags {
-    return Object.entries(tags).reduce((acc, [key, value]) => {
+  public async map(name: string, tags: Tags): Promise<Tags> {
+    const finalTagsObj: Tags = {};
+    const redisKeysArr: string[] = [];
+    const domainFields = await this.domainFieldsRepo.getDomainFieldsList();
+
+    Object.entries(tags).forEach(([key, value]) => {
       if (this.keyIgnoreSets[name].has(key)) {
-        return acc;
+        return;
+      }
+      if (domainFields.has(key.toUpperCase())) {
+        if (value !== null) {
+          redisKeysArr.push(`${key.toUpperCase()}:${value.toString()}`);
+        }
       }
-      return { ...acc, [`${name}_${key}`]: value };
-    }, {});
+      finalTagsObj[`${name}_${key}`] = value;
+    });
+
+    if (redisKeysArr.length > 0) {
+      const redisRes = await this.domainFieldsRepo.getFields(redisKeysArr);
+      redisRes.forEach((key, index) => {
+        if (key !== null) {
+          finalTagsObj[`${name}_${redisKeysArr[index].split(':')[0]}_DOMAIN`] = key;
+        }
+      });
+    }
+    return finalTagsObj;
```

Comment:
add some comments so it will be easier to understand the flow.

Reply:
done